server/src/main/java/com/hedera/block/server/pbj/PbjBlockStreamServiceProxy.java

```diff
@@ -0,0 +1,203 @@
+
+        final String observerClassType = blockNodeContext
+                .configuration()
+                .getConfigData(ProducerConfig.class)
+                .type();
+
+        if ("NOOP".equalsIgnoreCase(observerClassType)) {
+            // No need to register with the notifier for NOOP
+            return new NoOpProducerObserver(helidonProducerObserver, blockNodeContext);
+        } else {
+            final var producerBlockItemObserver = new ProducerBlockItemObserver(
+                    Clock.systemDefaultZone(),
+                    streamMediator,
+                    notifier,
+                    helidonProducerObserver,
+                    blockNodeContext,
+                    serviceStatus);
+
+            if (serviceStatus.isRunning()) {
+                // Register the producer observer with the notifier to publish responses back to the
+                // producer
+                notifier.subscribe(producerBlockItemObserver);
+            }
+
+            return producerBlockItemObserver;
+        }
+    }
+
+    void subscribeBlockStream(
+            SubscribeStreamRequest subscribeStreamRequest,
+            Flow.Subscriber<? super SubscribeStreamResponse> subscribeStreamResponseObserver) {
+
+        LOGGER.log(DEBUG, "Executing Server Streaming subscribeBlockStream gRPC method");
+
+        if (serviceStatus.isRunning()) {
+            // Unsubscribe any expired notifiers
+            streamMediator.unsubscribeAllExpired();
+
+            final var consumerStreamResponseObserver = new ConsumerStreamResponseObserver(
+                    Clock.systemDefaultZone(), streamMediator, subscribeStreamResponseObserver, blockNodeContext);
+
+            streamMediator.subscribe(consumerStreamResponseObserver);
+        } else {
+            LOGGER.log(ERROR, "Server Streaming subscribeBlockStream gRPC Service is not currently running");
+
+            subscribeStreamResponseObserver.onNext(SubscribeStreamResponse.newBuilder()
+                    .status(SubscribeStreamResponseCode.READ_STREAM_SUCCESS)
+                    .build());
+        }
+    }
+
+    @NonNull
+    private SubscribeStreamRequest parseSubscribeStreamRequest(
+            @NonNull final Bytes message, @NonNull final RequestOptions options) throws ParseException {
+        // Copying bytes to avoid using references passed from Helidon
+        return SubscribeStreamRequest.PROTOBUF.parse(Bytes.wrap(message.toByteArray()));
+    }
+
+    @NonNull
+    private Bytes createSubscribeStreamResponse(
+            @NonNull final SubscribeStreamResponse subscribeStreamResponse, @NonNull final RequestOptions options) {
+        return SubscribeStreamResponse.PROTOBUF.toBytes(subscribeStreamResponse);
+    }
+
+    @NonNull
+    private PublishStreamRequest parsePublishStreamRequest(
+            @NonNull final Bytes message, @NonNull final RequestOptions options) throws ParseException {
+        // Copying bytes to avoid using references from Helidon
```

Comment:
could we add here a TODO, to verify if is really necessary and if not to remove the copy bytes in a future PR?

Reply:
Good point. I'll add TODOs for the *Proxy files where applicable

Reply:
Done